### PR TITLE
implementation of moving pieces

### PR DESCRIPTION
**Pull request by RaykPlatzek**
1. created Piece show page
2. created a link for a sample piece on the game show page
3. made the link go to piece controller show
4. created a unique show page for every piece (avoiding browser cache issues by using Time.now.to_i)
5. made it possible to highlighted the sample piece and its destination field
6. created AJAX request to send destination row and column to piece controller and update column and row of piece in DB
7. redirect to game show page with updated piece coordinates

### CODE REVIEW

**Comment by RaykPlatzek** [comment]
We need to update the sample piece and make use of all pieces on the board once the board is populated with pieces

**Comment by mscoutermarsh** [comment]
After https://github.com/TheFirehoseProject-TeamProject/ChessApp/pull/20 is merged, will need to change a couple pieces of this (since drawing the pieces on board has changed a little bit). :smile:

Coordinate with @roderickjoseph 🙌

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+  def update
+    @piece = Piece.find(params[:id])
+    @piece.update(piece_params)
+    redirect_to game_path(@piece.game_id)
+  end
+
+  private
+
+  def current_game
+    @current_game ||= Game.find_by_id(@piece.game_id)
```
Comment (mscoutermarsh) [inline app/controllers/pieces_controller.rb]: can use `Game.find`, which is the same as `Game.find_by_id`.

File: app/views/pieces/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<div class="chessboard">
+  <% @board.each_with_index do |row, index_row| %>
+      <div class = "row offset-3">
+        <% row.each_with_index do |field, index_field| %>
+          <span class="chess_field <%=field[:class]%>" row="<%=index_row%>" column= "<%=index_field%>">
+            <%if index_field == @piece.column_coordinate && index_row == @piece.row_coordinate %>
+              <span><%= image_tag @piece.image, class: 'chess_piece_selected'%></span>
```
Comment (mscoutermarsh) [inline app/views/pieces/show.html.erb]: nice!

File: app/views/pieces/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      $("[row=" + "'" + row + "'" + "][column=" + "'" + column + "'" + "]").css("color","yellow");
+      $.post("/pieces/" + pieceId, {
+        _method: "PUT",
+        piece: {
+        row_coordinate: row,
+        column_coordinate: column
+        }
+      });
+    });
+  });
```
Comment (mscoutermarsh) [inline app/views/pieces/show.html.erb]: Looks good to me. This code will be useful later when implementing drag n drop.

**Pull request merged**